app_dart/lib/src/request_handlers/deflake_flaky_builders.dart

```diff
@@ -0,0 +1,139 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:github/github.dart';
+import 'package:meta/meta.dart';
+import 'package:yaml/yaml.dart';
+
+import '../request_handling/api_request_handler.dart';
+import '../request_handling/authentication.dart';
+import '../request_handling/body.dart';
+import '../service/bigquery.dart';
+import '../service/config.dart';
+import '../service/github_service.dart';
+import 'flaky_handler_utils.dart';
+
+@immutable
+class DeflakeFlakyBuilders extends ApiRequestHandler<Body> {
+  const DeflakeFlakyBuilders(Config config, AuthenticationProvider authenticationProvider)
+      : super(config: config, authenticationProvider: authenticationProvider);
+
+  static const int kRecordAmount = 50;
+
+  static final RegExp _issueLinkRegex = RegExp(r'https://github.com/flutter/flutter/issues/(?<id>[0-9]+)');
+
+  @override
+  Future<Body> get() async {
+    final RepositorySlug slug = config.flutterSlug;
+    final GithubService gitHub = config.createGithubServiceWithToken(await config.githubOAuthToken);
+    final BigqueryService bigquery = await config.createBigQueryService();
+    final String ciContent = await gitHub.getFileContent(slug, kCiYamlPath);
+    final List<_BuilderInfo> eligibleBuilders = await _getEligibleFlakyBuilders(gitHub, slug, content: ciContent);
+    String testOwnerContent;
+    for (final _BuilderInfo info in eligibleBuilders) {
+      final List<BuilderRecord> builderRecords =
+          await bigquery.listRecentBuildRecordsForBuilder(kBigQueryProjectId, builder: info.name, limit: kRecordAmount);
+      if (builderRecords.every((BuilderRecord record) => !record.isFlaky)) {
+        testOwnerContent ??= await gitHub.getFileContent(slug, kTestOwnerPath);
+        await _fileDeflakePullRequest(gitHub, slug,
+            info: info, ciContent: ciContent, testOwnerContent: testOwnerContent);
+      }
+    }
+    return Body.forJson(const <String, dynamic>{
+      'Status': 'success',
+    });
+  }
+
+  Future<List<_BuilderInfo>> _getEligibleFlakyBuilders(GithubService gitHub, RepositorySlug slug,
+      {String content}) async {
+    final YamlMap ci = loadYaml(content) as YamlMap;
+    final YamlList targets = ci[kCiYamlTargetsKey] as YamlList;
+    final List<YamlMap> flakyTargets = targets
+        .where((dynamic target) => target[kCiYamlTargetIsFlakyKey] == true)
+        .map<YamlMap>((dynamic target) => target as YamlMap)
+        .toList();
+    final List<_BuilderInfo> result = <_BuilderInfo>[];
+    final List<String> lines = content.split('\n');
+    final Map<String, PullRequest> nameToExistingPRs = await getExistingPRs(gitHub, slug);
+    for (final YamlMap flakyTarget in flakyTargets) {
```

Comment:
As we talked about, please add a list to exclude builders that we want to keep them as flaky all the time, like the ios32 ones.

Reply:
Do you have a list?

Reply:
At least these two for now: `Mac_ios flutter_gallery__transition_perf_e2e_ios32` and `Mac_ios native_ui_tests_ios32`.